src/Sticky.jsx

```diff
@@ -380,10 +380,12 @@ class Sticky extends Component {
             [this.props.releasedClass]: this.state.status === STATUS_RELEASED
         })
 
+        var children = this.props.children;
+
         return (
             <div ref={(outer) => { this.outerElement = outer; }} className={outerClasses} style={outerStyle}>
                 <div ref={(inner) => { this.innerElement = inner; }} className='sticky-inner-wrapper' style={innerStyle}>
-                    {this.props.children}
+                    {typeof children === 'function' ? children({ status: this.state.status }) : children}
```

Comment:
Would it make sense to pass the `state` into the children instead of the `status` only?

Reply:
Not sure other things from the component state should be available to children, also `onStateChange` passes `status` only too https://github.com/yahoo/react-stickynode/blob/master/src/Sticky.jsx#L290-L293